users/vieting/jobs/fairseq_labels.py

```diff
@@ -0,0 +1,252 @@
+        
+                if rand.random() > self.valid_percent:
+                    train_data.append((rel_path, frames))
+                else:
+                    print(f"{rel_path}\t{frames}", file=valid_f)
+            all_train_data.append(train_data)
+
+        if valid_f is not None:
+            valid_f.close()
+
+        #with open(self.out_train_tsv_path.get(), "w") as f:
+        with open(self.out_train_tsv_path, "w") as f:
+            print(common_dir, file=f)
+            for train_data in all_train_data:
+                for rel_path, frames in train_data:
+                    print(f"{rel_path}\t{frames}", file=f)
+        
+
+        
+    def create_labels(self, split=None):
+        """
+        Creates the labels for fairseq wav2vec finetuning (train.ltr, train.wrd, valid.ltr, valid.wrd).
+
+        :param str split: either "train" or "valid"
+        """
+        assert split in ["train", "valid"], f"split must be either 'train' or 'valid', but is {split}"
+
+
+        transcriptions = {}
+        if split == "train":
+            tsv_path = self.out_train_tsv_path.get()
+            ltr_path = self.out_train_ltr_path.get()
+            wrd_path = self.out_train_wrd_path.get()
+        else:
+            tsv_path = self.out_valid_tsv_path.get()
+            ltr_path = self.out_valid_ltr_path.get()
+            wrd_path = self.out_valid_wrd_path.get()
+
+        with open(tsv_path, "r") as tsv, open(
+            ltr_path, "w"
+        ) as ltr_out, open(
+            wrd_path, "w"
+        ) as wrd_out:
+            root = next(tsv).strip()
+            print('root',root)
+            for line in tsv:
+                line = line.strip()
+
+                dir = os.path.dirname(line)
+
+                if dir not in transcriptions:
+                    parts = dir.split(os.path.sep)
+
+                    trans_path = f"{os.path.join()}"
+
+                    path = os.path.join(root, dir, "transcriptions.txt")
+
+                    assert os.path.exists(path)
+                    texts = {}
+                    with open(path, "r") as trans_f:
+                        for tline in trans_f:
+                            items = tline.strip().split()
+                            texts[items[0]] = " ".join(items[1:])
+
+                    transcriptions[dir] = texts
+                part = os.path.basename(line).split(".")[0]+f'.{self.file_extension}'
+
+                assert part in transcriptions[dir]
+                print(transcriptions[dir][part], file=wrd_out)
+                print(
+                    " ".join(list(transcriptions[dir][part].replace(" ", "|"))) + " |",
+                    file=ltr_out,
+                )
+
+
+    def download_dict_ltr(self):
+        """
+        Downloads the dict.ltr.txt file for fairseq wav2vec finetuning.
+        """
+        subprocess.check_call(["wget", "-O", self.out_dict_ltr_path.get(), "https://dl.fbaipublicfiles.com/fairseq/wav2vec/dict.ltr.txt"])
```

Comment:
Since it's just a rather small file, I'd prefer to store it as a string here and print it into a file rather than relying on this link which might be changed or removed in the future.